velox/vector/arrow/Bridge.cpp

```diff
@@ -1518,8 +1518,18 @@ VectorPtr importFromArrowImpl(
         arrowArray.n_buffers,
         2,
         "Primitive types expect two buffers as input.");
-    auto values = wrapInBufferView(
-        arrowArray.buffers[1], arrowArray.length * type->cppSizeInBytes());
+    BufferPtr values = nullptr;
+    if (type->isShortDecimal()) {
+      values = AlignedBuffer::allocate<int64_t>(arrowArray.length, pool);
+      auto rawValues = values->asMutable<int64_t>();
+      auto oldValues = static_cast<const int128_t*>(arrowArray.buffers[1]);
```

Comment:
It looks like arrow array can hold a decimal of a bit-width other than int128_t (int256 is also possible). Maybe we can add a check in Velox [importFromArrowImpl](https://github.com/facebookincubator/velox/blob/main/velox/vector/arrow/Bridge.cpp#L969-L982) to ensure the bit-width is not set or the bit-width is indeed 128. Then it should be safe to case buffers[1] as int128* here.
https://github.com/apache/arrow/blob/main/cpp/src/arrow/c/bridge.cc#L348-L356

Reply:
Since type is firstly called here: https://github.com/facebookincubator/velox/blob/09ba2205b84d0b9ff27cf2998a860636420b5385/velox/vector/arrow/Bridge.cpp#L1443, whereas the decimal type must be `int128_t`, otherwise [importFromArrowImpl](https://github.com/facebookincubator/velox/blob/main/velox/vector/arrow/Bridge.cpp#L969-L982) could already fail, do we need to add extra check here?

Reply:
I tried below cases, and they both work. Could you help confirm?

> EXPECT_EQ(*DECIMAL(10, 4), *testSchemaImport("d:10,4,128"));
  EXPECT_EQ(*DECIMAL(10, 4), *testSchemaImport("d:10,4,256"));